modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/files_query.rb

```diff
@@ -0,0 +1,124 @@
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module Peripherals
+    module StorageInteraction
+      module OneDrive
+        class FilesQuery
+          FIELDS = "?$select=id,name,size,webUrl,lastModifiedBy,createdBy,fileSystemInfo,file,folder"
+
+          using ServiceResultRefinements
+          def self.call(storage:, user:, folder:)
+            new(storage).call(user:, folder:)
+          end
+
+          def initialize(storage)
+            @storage = storage
+          end
+
+          def call(user:, folder: nil)
+            result = using_user_token(user) do |token|
+              # Make the Get Request to the necessary endpoints
+              response = Net::HTTP.start(GRAPH_API_URI.host, GRAPH_API_URI.port, use_ssl: true) do |http|
+                http.get(uri_path_for(folder) + FIELDS, { 'Authorization' => "Bearer #{token.access_token}" })
+              end
+
+              handle_response(response)
+            end
+
+            result.map { |json_files| storage_files(json_files) }
+          end
+
+          private
+
+          def handle_response(response)
+            case response
+            when Net::HTTPSuccess
+              ServiceResult.success(result: MultiJson.load(response.body, symbolize_keys: true)[:value])
+            when Net::HTTPNotFound
+              ServiceResult.failure(result: :not_found, errors: ::Storages::StorageError.new(code: :not_found))
+            when Net::HTTPUnauthorized
+              ServiceResult.failure(result: :not_authorized, errors: ::Storages::StorageError.new(code: :not_authorized))
+            else
+              ServiceResult.failure(result: :error, errors: ::Storages::StorageError.new(code: :error))
+            end
+          end
+
+          def uri_path_for(folder)
+            return "/v1.0/me/drive/root/children" unless folder
+
+            "/v1.0/drives/#{@storage.drive_id}/items/#{folder}/children"
+          end
+
+          def storage_files(json_files)
+            json_files.map do |json|
+              StorageFile.new(
+                id: json[:id],
+                name: json[:name],
+                size: json[:size],
+                mime_type: json.dig(:file, :mimeType) || 'application/x-op-directory',
+                created_at: DateTime.parse(json.dig(:fileSystemInfo, :createdDateTime)),
+                last_modified_at: DateTime.parse(json.dig(:fileSystemInfo, :lastModifiedDateTime)),
+                created_by_name: json.dig(:createdBy, :user, :displayName),
+                last_modified_by_name: json.dig(:lastModifiedBy, :user, :displayName),
+                location: json[:webUrl],
+                permissions: nil
+              )
+            end
+          end
+
+          def mime_type(json)
+            json.dig(:file, :mimeType) || json.key?(:folder) ? 'application/x-op-directory' : nil
+          end
+
+          def using_user_token(user, &block)
```

Comment:
🟢 Maybe we can figure something out with the `Registry` to instantiate the different kinds of `ConnectionManager`s. But let's see, what a discussion about the removal of these class will bring.